sdcm/cluster.py

```diff
@@ -3610,18 +3610,23 @@ def collect_logs(self, storing_dir):
 
         return storing_dir
 
-    def get_prometheus_snapshot(self, node):
-        # avoid cyclic-decencies between cluster and db_stats
-
+    @retrying(n=3, sleep_time=2, allowed_exceptions=(PrometheusSnapshotErrorException, ), message='Create prometheus snapshot')
+    def request_prometheus_snapshot(self, node):
         ps = PrometheusDBStats(host=node.public_ip_address)
         result = ps.create_snapshot()
-        self.log.debug(result)
         if "success" in result['status']:
             snapshot_dir = os.path.join(self.monitoring_data_dir,
                                         "snapshots",
                                         result['data']['name'])
+            return snapshot_dir
         else:
-            return ""
+            raise PrometheusSnapshotErrorException
+
+    def get_prometheus_snapshot(self, node):
+        try:
+            snapshot_dir = self.request_prometheus_snapshot(node)
+        except PrometheusSnapshotErrorException:
+            snapshot_dir = self.monitoring_data_dir
```

Comment:
add a warning to log that you are failing back to the data dir

Reply:
fixed